target/scripts/build/packages.sh

```diff
@@ -18,16 +21,44 @@ function _pre_installation_steps() {
   apt-get "${QUIET}" update
 
   _log 'trace' 'Installing packages that are needed early'
-  apt-get "${QUIET}" install --no-install-recommends apt-utils 2>/dev/null
+  # add packages usually required by apt to
+  # - not log unnecessary warnings
+  # - be able to add PPAs early (e.g., Rspamd)
+  local EARLY_PACKAGES=(
+    apt-utils # avoid useless warnings
+    apt-transport-https ca-certificates curl gnupg # required for adding PPAs
+    systemd-standalone-sysusers # avoid problems with SA / Amavis (https://github.com/docker-mailserver/docker-mailserver/pull/3403#pullrequestreview-1596689953)
+  )
+  apt-get "${QUIET}" install --no-install-recommends "${EARLY_PACKAGES[@]}" 2>/dev/null
+
+  _log 'trace' 'Adding Rspamd PPA'
+  curl -sSfL https://rspamd.com/apt-stable/gpg.key | gpg --dearmor >/etc/apt/trusted.gpg.d/rspamd.gpg
+  echo "deb [signed-by=/etc/apt/trusted.gpg.d/rspamd.gpg] http://rspamd.com/apt-stable/ ${VERSION_CODENAME} main" >/etc/apt/sources.list.d/rspamd.list
+
+  _log 'trace' 'Updating package index after adding PPAs'
+  apt-get "${QUIET}" update
```

Comment:
Why did you move the rspam part here?

IMO the old function `_install_rspamd()` was better. All rspamd stuff in one function + that maked it clear, that rspam is not installed from the debian repo.

Reply:
This was [raised as a concern during my initial review](https://github.com/docker-mailserver/docker-mailserver/pull/3403#discussion_r1306372918) as well with @georglauterbach justifying it:

> _I believe the new spot to be more appropriate as the custom Rspamd installation function has been removed._

I too prefer keeping features better isolated and see no need to bundle them all into the same package install, although we could probably do that if we export an array variable that can be appended to by separate feature functions? 🤔

Reply:
1. You forgot to add `_add_ppas` at the bottom of the file
2. IMO the function name should be more clear: `_add_rspamd_ppas`
    1. Or still my personal favorit: `_install_rspamd` with `apt-get "${QUIET}" --no-install-recommends install 'rspamd' 'redis-server'` added.

Reply:
Right, the line should be added indeed. Having the installation routine later install Rspamd makes the whole thing faster, though: we do not require a separate `apt-get install` and all PPAs that we potentially use in the future can be added to this method (hence the name; not being specific to Rspamd was my initial intention). What do you think?